docs/contributor/06-module-migration-guide.md

```diff
@@ -0,0 +1,231 @@
+# Module Migration Guideline
+
+This guide provides detailed instructions for how to migrate a Kyma module from existing (old) module metadata, especially channel-based ModuleTemplates, to the new module metadata, i.e. version-based ModuleTemplates accompanied by ModuleReleaseMeta.
+
+It is highly recommended that module teams familiarize themselves with the new module metadata before starting the migration:
+- The ADR backing the migration is [#984](https://github.com/kyma-project/community/issues/984)
+- An Update on Module Metadata has also been given in the 2024-11-26 Iteration Review
```

Comment:
I could not find any recording for that Iteration review, it might have expired... See [Recordings](https://sap-my.sharepoint.com/personal/varban_vasilev_sap_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fvarban%5Fvasilev%5Fsap%5Fcom%2FDocuments%2FRecordings&FolderCTID=0x01200043C05106F771C94CA366B210D0FB41BF&view=0). I don't know if it helps... maybe the [Presentation](https://sap-my.sharepoint.com/:p:/p/c_schwaegerl/EbvSNmRnr3JEjaLoZ__cI9UB9lu5tt0qaly-f7yQO2Gwbw?e=slfiDf) linked in the IR Wiki would be better? Or we could create a doc comparing old and new metadata and link it here.